ablateLibrary/particles/particles.cpp

```diff
@@ -544,25 +544,118 @@ static PetscErrorCode DMSequenceViewTimeHDF5(DM dm, PetscViewer viewer) {
     PetscFunctionReturn(0);
 }
 
-void ablate::particles::Particles::View(PetscViewer viewer, PetscInt steps, PetscReal time, Vec u) const {
+void ablate::particles::Particles::Save(PetscViewer viewer, PetscInt steps, PetscReal time) const {
     DMSetOutputSequenceNumber(GetDM(), steps, time) >> checkError;
-
-    PetscFunctionBegin;
     Vec particleVector;
 
     for (auto const &field : particleFieldDescriptors) {
-        if (field.type == PETSC_DOUBLE) {
+        if (field.type == PETSC_REAL) {
             DMSwarmCreateGlobalVectorFromField(GetDM(), field.fieldName.c_str(), &particleVector) >> checkError;
             PetscObjectSetName((PetscObject)particleVector, field.fieldName.c_str()) >> checkError;
             VecView(particleVector, viewer) >> checkError;
             DMSwarmDestroyGlobalVectorFromField(GetDM(), field.fieldName.c_str(), &particleVector) >> checkError;
         }
     }
 
+    // Get the particle info
+    int rank;
+    MPI_Comm_rank(PetscObjectComm((PetscObject)GetDM()), &rank) >> checkMpiError;
+
+    // get the local number of particles
+    PetscInt globalSize;
```

Comment:
@dsalac These are changes we talked about today.  I'm saving the global size of the dmSwarm and then restoring the particles "evenly" on resume. There is no assumption on save/restore ranks.